common/rpc/encryption/tlsFactory.go

```diff
@@ -59,16 +59,17 @@ const (
 )
 
 // NewTLSConfigProviderFromConfig creates a new TLS Configuration provider from RootTLS config
-func NewTLSConfigProviderFromConfig(encryptionSettings *config.RootTLS, hostname string) (TLSConfigProvider, error) {
-	if encryptionSettings == nil /*|| encryptionSettings.Provider == "" */ {
+func NewTLSConfigProviderFromConfig(encryptionSettings config.RootTLS, hostname string) (TLSConfigProvider, error) {
+	/* if || encryptionSettings.Provider == ""  {
```

Comment:
it looks like this function could be cleaned up a bit.

Reply:
Leaving for now, plan to come back and refactor this and commented code might become live, but will clean up in next iteration either way.